docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/eksctl.md

```diff
@@ -0,0 +1,288 @@
+
+## Prerequisites
+
+- an [AWS account](https://docs.aws.amazon.com/accounts/latest/reference/accounts-welcome.html) is required to create any resources within AWS.
+- [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html) is a CLI tool for creating AWS resources.
+- [eksctl](https://eksctl.io/installation/) is a CLI tool for creating and managing EKS clusters.
+- [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) to interact with the cluster.
+
+## Considerations
+
+The following does not reflect a production ready setup but is a good quick-start to get going with Camunda on AWS EKS.
+
+For the simplicity of this guide we'll be using non best practices but refer to further documents to allow you to delve deeper into the topic.
+
+## Outcome
+
+Following this guide will result in
+
+- an AWS EKS Kubernetes Cluster with 4 nodes and the possibility to scale up further.
+- the [EBS CSI driver](https://docs.aws.amazon.com/eks/latest/userguide/ebs-csi.html) is installed and configured.
+- a [managed PostgreSQL](https://aws.amazon.com/rds/postgresql/) instance is spawned that will be used by the Camunda applications.
+
+This basic cluster setup is required to continue with the Helm setup as described in our [AWS Helm Guide](#). <!-- TODO: reference future guide -->
+
+## Usage
+
+The eksctl tool allows creating clusters via a single command but this doesn't support all configuration options. Therefore, we're supplying a `yaml` file that can be used with the CLI to create the cluster preconfigured with various settings.
+
+### Eksctl Prerequisites
+
+Setup authentication to allow interacting with `AWS` via the [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html) to configure access.
+
+General advice, the user creating the resources will be the owner of those and always linked to those. Meaning the user will always have admin access on Kubernetes, except you delete it. Therefore, it can make sense to create an extra [IAM user](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users.html) that's solely used for the EKSCTL purposes.
+
+<!--
+```shell
+eksctl create cluster --name test-cluster --region eu-central-1 --with-oidc --version 1.28 --nodegroup-name services --instance-types m5.xlarge --nodes 4 --nodes-min 1 --nodes-max 10 --node-private-networking --disable-pod-imds --vpc-cidr 10.192.0.0/16 --vpc-nat-mode HighlyAvailable --dry-run
+```
+-->
+
+### Eksctl Cluster yaml
+
+The following `yaml` has to be copied to a file on your system and can be executed with.
+
+```shell
+eksctl create cluster --config-file cluster.yaml
+```
+
+Where `cluster.yaml` contains the following contents.
+Before applying, please adjust the `<region>` and `<zone>` and rename the cluster name with the placeholder `<clusterName>` to your liking.
+
+```yaml
```

Comment:
This would also create somewhat an excplicit interface for the user to customize the parameters of their cluster.

Reply:
should be possible as Docusaurus allows writing React components, so we can have a bit more interactive and embedded UI.
I'd suggest creating a ticket so we don't lose track of the idea.

I'm not sure it's a good idea to go with any sort of script as it should be OS agnostic and Windows is quite common.
The CLIs are fine since they work the same way.

Reply:
Fair point about Windows users. The variable substitution looks same in Powershell and Shell it [seems](https://learn.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-string-substitutions?view=powershell-7.3#variable-substitution), maybe that just using the variables is sufficient.

On the cross-platform approach, we can also consider using Docusaurus content tabs and provide 2 flavours of commands for users to switch between (Unix vs Windows).

Reply:
For the first iteration, I'd suggest we concentrate on UNIX with a disclaimer in the consideration that it can be easily adapted to Windows.
In a future iteration, we either go the react way or use tabs. Of which I would find react easier since we don't have to continuously test it on windows.